### ISSUE TITLE
block size 16384 for XZ makes compression pointless

### ISSUE THREAD

**Issue by develar** [AppImage/AppImageKit#697]
If user decided to use XZ compression instead of default gzip — he really wants small size of AppImage. Using `16384` as a block size makes XZ compression like gzip — for example, for Electron App 51.9 vs 50.4.

So, electron-builder now uses `1048576` instead of  `16384`. Size difference — 50.4 vs 39.1. 11.3 MB difference. Slow start-up time? Well, it is your choice to use XZ.

I am not going to convince you to change this setting in the AppImageTool, JFYI.

**Comment by probonopd** [comment]
@develar we are open to discussing the defaults, but the current defaults are what proved to be best in our test of various apps (including LibreOffice) some time ago.

Could it be that we are seeing effects of the asar compression in Electron builds? It is probably not helpful for size nor speed to compress files in an asar and then compress the asar in a squashfs file. Is there a possibility to ship the contents of the asar uncompressed?

**Comment by develar** [comment]
> slower application startup times.

Yes, it is a reason why electron-builder doesn't use XZ by default. My point is — if user decided to use XZ, current very small block size leads to strange results (final size ~= ZIP size). The same situation if you will use ZSTD stream API incorrectly (for ZSTD minimal required buffer size = 1MB (better 32MB), in case of squashfs we cannot experiment — 1MB it is maximum).

> a possibility to ship the contents of the asar uncompressed

User can, but it will be never default for electron-builder.  Problem — if you use normal OS (macOS or Linux), you don't need to use ASAR, but still you have to provide windows version of app. And on Windows you should use ASAR. So, if different settings (asar true on win and false otherwise) will be applied, it will lead to situation when on release day you will discover that your app doesn't work well on Windows (because ASAR has some limitations and pitfalls). And it is not acceptable.

**Comment by probonopd** [comment]
> So, if different settings (asar true on win and false otherwise) will be applied, it will lead to situation when on release day you will discover that your app doesn't work well on Windows (because ASAR has some limitations and pitfalls). And it is not acceptable.

Interesting, I understand. Thanks for the explanation.

**Comment by ArtemAvramenko** [comment]
I can't imagine a case where xz compression would be acceptable for an Electron.js app.

When compression is set to "maximum" any other algorithm (lzo, lz4 or zstd) should be used. Let there be a slightly larger output size, since we have to be satisfied with the "normal" level at the moment.

Of course you need to decide compatibility with different kernels, so for example zstd may not be the best option.
